### PR TITLE
Feature/dark mode

### LINKED ISSUE #169: Dark Mode Support
One difference from running both the German Corona Warn and SwissCovid App I noticed the currently missing dark mode in SwissCovid. Is this in the Roadmap or considered out of scope? Design wise MeteoSwiss recently implemented a dark mode with a similar design language as SwissCovid.

### PR DESCRIPTION
**Pull request by alexandru-lazar**
Closes #169

### CODE REVIEW

**Comment by alexandru-lazar** [comment]
#169

**Comment by joseph-elmallah** [comment]
Some remarks while testing the branch:
- During the onboarding on the Tips on use page, transitioning to Dark mode leave the bottom two images in black. They should flip to white. Best to achieve this using template and tint colour, if not possible we can export a set of white images.
- In the entering a COVID code page, the disabled button state is shown as white in dark mode, which can lead into people thinking it can be interacted with, before filling all digits. **NSButton.swift** line 121 the background is set always to fixed black transparent. I checked with the design team and it should be **#cdcdd0** in dark mode.
- The impressum is still all white. We need to provide you with a new CSS that support the new Apple attributes for dark mode since the impressum is made with a WKWebView.

**Comment by alexandru-lazar** [comment]
Implemented the first two remarks and I want to add another minor issue:
Simulator Screen Shot - iPhone 11 Pro - 2020-07-15 at 13 43 23
In dark mode the bottom image is not displayed correctly. If I use the rendering mode and change the tint colour the red icon will have the tint colour. We should have a new image there for dark mode. Awaiting new CSS and this image to finish the implementation. Thank you for your feedback.

**Comment by joseph-elmallah** [comment]
Please find here the BAG logo in dark mode.
[Archive.zip](https://github.com/DP-3T/dp3t-app-ios-ch/files/4930468/Archive.zip)

**Comment by joseph-elmallah** [comment]
Attached you can find the full impressum with dark mode support. Please replace the folder structure with the unzipped version of the folder.
[impressum.zip](https://github.com/DP-3T/dp3t-app-ios-ch/files/4931855/impressum.zip)

File: DP3TApp/SharedUI/Style/UIColor+NS.swift [hunk 1]
```diff
@@ -11,23 +11,60 @@
...
+    
+    static var ns_backgroundTertiary: UIColor {
+        return UIColor().setColorsForTheme(lightColor: UIColor(ub_hexString: "#efefef")!, darkColor: UIColor(ub_hexString: "#1E1E23")!)
+    }
+    
+    static var disabledButtonBackground: UIColor {
+        return UIColor().setColorsForTheme(lightColor: UIColor.black.withAlphaComponent(0.15), darkColor: UIColor(ub_hexString: "#cdcdd0")!)
+    }
+    
+    static var imageTintColor: UIColor {
```
Comment (joseph-elmallah) [inline DP3TApp/SharedUI/Style/UIColor+NS.swift:61]: Naming comment: **imageTintColor** is a bit vague and would be better understood if it was more specific. I would suggest naming it ns_disclaimer_icon_color

File: DP3TApp/SharedUI/Style/UIColor+NS.swift [hunk 2]
```diff
@@ -11,23 +11,60 @@
...
-
-    public static var ns_background = UIColor.white
-    public static var ns_backgroundSecondary = UIColor(ub_hexString: "#f7f7f7")!
-    public static var ns_backgroundTertiary = UIColor(ub_hexString: "#efefef")!
+    
+    static var ns_background: UIColor {
+        return UIColor().setColorsForTheme(lightColor: UIColor.white, darkColor: UIColor.black)
+    }
+    
+    static var moduleBackground: UIColor {
```
Comment (joseph-elmallah) [inline DP3TApp/SharedUI/Style/UIColor+NS.swift:45]: To keep stuff consistent this needs to be renamed to ns_moduleBackground

File: DP3TApp/SharedUI/Style/UIColor+NS.swift [hunk 3]
```diff
@@ -11,23 +11,60 @@
...
+    
+    static var ns_backgroundSecondary: UIColor {
+        return UIColor().setColorsForTheme(lightColor: UIColor(ub_hexString: "#f7f7f7")!, darkColor: UIColor.black)
+    }
+    
+    static var ns_backgroundTertiary: UIColor {
+        return UIColor().setColorsForTheme(lightColor: UIColor(ub_hexString: "#efefef")!, darkColor: UIColor(ub_hexString: "#1E1E23")!)
+    }
+    
+    static var disabledButtonBackground: UIColor {
```
Comment (joseph-elmallah) [inline DP3TApp/SharedUI/Style/UIColor+NS.swift:57]: To keep stuff consistent this needs to be renamed to ns_disabledButtonBackground

File: DP3TApp/SharedUI/Style/UIColor+NS.swift [hunk 4]
```diff
@@ -11,23 +11,60 @@
...
+        return UIColor().setColorsForTheme(lightColor: UIColor(ub_hexString: "#e20008")!, darkColor: UIColor(ub_hexString: "#FF3B42")!)
+    }
+    
+    // MARK: - Text color
 
+    public static var ns_text: UIColor {
+        return UIColor().setColorsForTheme(lightColor: UIColor.defaultTextColor, darkColor: UIColor.white.withHighContrastColor(color: .black))
+    }
+    
+    public static var defaultTextColor = UIColor.ns_darkBlueBackground.withHighContrastColor(color: .black)
```
Comment (joseph-elmallah) [inline DP3TApp/SharedUI/Style/UIColor+NS.swift:25]: To keep stuff consistent this needs to be renamed to ns_defaultTextColor

**Comment by joseph-elmallah** [comment]
@alexandru-lazar We will merge the pull request first in a branch so we can make a build and give it for UX review, if all goes well we will merge it back to develop. Thanks for your help.

**Pull request merged**